pandas/core/internals.py

```diff
@@ -1394,6 +1405,9 @@ def where(self, other, cond, align=True, raise_on_error=True,
         if not hasattr(cond, 'shape'):
             raise ValueError("where must have a condition that is ndarray "
                              "like")
+        else:
```

Comment:
huh? if you find yourself adding an if/then pretty much anywhere then you are doing it wrong. better to add the method to Sparse and call super; sometimes the super method may need a bit of refactor to make it more general.